pkg/apis/servicecatalog/v1beta1/class.go

```diff
@@ -55,3 +60,39 @@ func (c *ClusterServiceClass) GetDescription() string {
 func (c *ServiceClass) GetDescription() string {
 	return c.Spec.Description
 }
+
+// GetSpec returns the spec for the class.
+func (c *ServiceClass) GetSpec() CommonServiceClassSpec {
+	return c.Spec.CommonServiceClassSpec
+}
+
+// GetSpec returns the spec for the class.
+func (c *ClusterServiceClass) GetSpec() CommonServiceClassSpec {
+	return c.Spec.CommonServiceClassSpec
+}
+
+// GetServiceBroker returns the name of the service broker for the class.
+func (c *ServiceClass) GetServiceBroker() string {
+	return c.Spec.ServiceBrokerName
+}
+
+// GetServiceBroker returns the name of the service broker for the class.
+func (c *ClusterServiceClass) GetServiceBroker() string {
+	return c.Spec.ClusterServiceBrokerName
+}
+
+// GetStatusText returns the sttaus of the class.
+func (c *ServiceClass) GetStatusText() string {
+	if c.Status.RemovedFromBrokerCatalog {
+		return statusDeprecated
+	}
+	return statusActive
+}
+
+// GetStatusText returns the sttaus of the class.
+func (c *ClusterServiceClass) GetStatusText() string {
```

Comment:
Both the functions responsible for getting the status have the same code, not sure if we can do with just one of them?

Reply:
Yes, you could define `GetStatusText()` on `CommonServiceClassStatus` and then return `c.Status.GetStatusText()` from these methods.

Reply:
Leaving it as it is for now

Reply:
+1 for adding CommonServiceClassStatus.GetStatusText() so as to remove duplicate code